femtools/Particles.F90

```diff
@@ -856,14 +978,13 @@ subroutine create_single_particle(detector_list, xfield, position, id, proc_id,
     call insert(detector, detector_list)
 
     ! Populate particle
-    detector%name = name
     detector%position = position
     detector%element = element
     detector%local_coords = lcoords
-    detector%type = LAGRANGIAN_DETECTOR
     detector%id_number = id
     detector%proc_id = proc_id
-    detector_list%proc_part_count = detector_list%proc_part_count + 1
+    detector%list_id = list_id
```

Comment:
Can we get this from `detector_list%id`, and avoid having to pass the list ID through so many levels (it's a bit abstraction-breaking too)?